network_diffusion/tpn/functions.py

```diff
@@ -0,0 +1,23 @@
+
+from typing import List
+
+import networkx as nx
+from networkx.algorithms.approximation.dominating_set import min_weighted_dominating_set
+from networkx.algorithms.dominating import dominating_set
+
+from network_diffusion.mln.mlnetwork import MultilayerNetwork
+from network_diffusion.tpn.tpnetwork import TemporalNetwork
+
+def driver_nodes(net: MultilayerNetwork) -> List[int]:
```

Comment:
This works only on single-layered network. Since models should work by default on multilayered graphs, i'd extend this method to work on multilayered graphs or at least raised an exception if provided network has more than one layer

Reply:
I have added support for multi-layered networks.